### ISSUE TITLE
Issue with rules configuration after updating to 0.54.0

### ISSUE THREAD

**Issue by Mucha92** [realm/SwiftLint#5347]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

After updating to version 0.54.0 a following problem appears:
```
[...]
warning: Invalid configuration for 'type_name'. Falling back to default.
[...]
ApplicationUtils.swift:34:12: error: Type Name Violation: Type name 'MIME_TYPES' should only contain alphanumeric and other allowed characters (type_name)
```

My configuration file (swiftlint.yml) contains the following lines:
```
type_name:
  allowed_symbols: "_"
  min_length: 3
  severity: warning
```

Did somethign change? Should I somehow update my configuration file? On 0.53.0 works fine.

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? **0.54.0**
* Installation method used (Homebrew, CocoaPods, building from source, etc)? **CocoaPods**
* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents. **No**
* Which Xcode version are you using (check `xcodebuild -version`)? **Xcode 14.2 Build version 14C18**

**Comment by jessesquires** [comment]
I think you want this:

```
identifier_name:
   allowed_symbols: ['_']
   min_length: 3
   severity: warning
```

**Comment by Mucha92** [comment]
Unfortunately it didn't help :(

Here's the simplified example of problem reproduction:

```
.
├─ test.swift
├─ .swiftlint.yml
├─ 53
│   └─ swiftlint
└─ 54
    └─ swiftlint
```

test.swift:
```swift 
class TestClass {
    struct ALLOWED_NAME {
    }
}

```
.swiftlint.yml:
```yml
identifier_name:
  allowed_symbols: ['_']
  min_length: 3
  max_length: 50
  severity: warning

type_name:
  allowed_symbols: ['_']
  min_length: 3
  severity: warning
```

result for 0.53.0:
```
./53/swiftlint version && ./53/swiftlint --config .swiftlint.yml
0.53.0
Linting Swift files in current working directory
Linting 'test.swift' (1/1)
Done linting! Found 0 violations, 0 serious in 1 file.
```

result for 0.54.0:
```
./54/swiftlint version && ./54/swiftlint --config .swiftlint.yml
0.54.0
warning: Invalid configuration for 'type_name'. Falling back to default.
Linting Swift files in current working directory
Linting 'test.swift' (1/1)
[...]/test.swift:2:12: error: Type Name Violation: Type name 'ALLOWED_NAME' should only contain alphanumeric and other allowed characters (type_name)
Done linting! Found 1 violation, 1 serious in 1 file.
```

I've noticed that if I change test.swift to:
```
class TestClass {
    let ALLOWED_NAME = 1
}

```
it works fine for both versions. So the issue seems to be related to interpretation of struct keyword

**Comment by Rvaler** [comment]
I'm facing a similar issue with the `nesting` property. This is my setup:

```
nesting:
  type_level:
    warning: 2
  statement_level:
    warning: 10
```

But when running `swiflint rules nesting` I'm getting the following message: 
```
warning: Invalid configuration for 'nesting'. Falling back to default.
Nesting (nesting): Types should be nested at most 1 level deep, and functions should be nested at most 2 levels deep.
```

**Comment by SimplyDanny** [comment]
@Mucha92: In your configuration, `severity` is an invalid key. The severity is specified via `min_length` and `max_length` like:

```yml
min_length:
  warning: 3
  error: 2
```

Using `min_length: 3` only configures the warning level.

**Comment by Mucha92** [comment]
Thank you, it solves problem
